### PR TITLE
add write concurrency parameter

### PR DESCRIPTION
**Pull request by marsishandsome**
close https://github.com/pingcap/tispark/issues/914

### CODE REVIEW

**Comment by marsishandsome** [comment]
> Mostly LGTM except PR description does not agree with the code.

cause this PR is based on https://github.com/pingcap/tispark/pull/913

File: core/src/main/scala/com/pingcap/tispark/TiBatchWrite.scala [hunk 1]
```diff
@@ -765,7 +765,7 @@ class TiRegionPartitioner(regions: List[TiRegion]) extends Partitioner {
       val region = regions(i)
       val range = KeyRangeUtils.makeRange(region.getStartKey, region.getEndKey)
       if (range.contains(rawKey)) {
-        return i
+        return if (writeConcurrency <= 0) i else i % writeConcurrency
```
Comment (birdstorm) [inline core/src/main/scala/com/pingcap/tispark/TiBatchWrite.scala]: why not `return i % numPartitions`

Comment (marsishandsome) [reply core/src/main/scala/com/pingcap/tispark/TiBatchWrite.scala]: good point

**Pull request merged**